### PR TITLE
RDKB-52790 : Webconfig is ignoring the WAN status events and configur…

### PR DESCRIPTION
**Pull request by PalakshaGowda**
…ations are not in sync during WAN start/stop

### CODE REVIEW

File: src/webcfg_wanhandle.c [hunk 1]
```diff
@@ -18,6 +18,103 @@
 
 #include "webcfg_wanhandle.h"
 
+#ifdef _SKY_HUB_COMMON_PRODUCT_REQ_
+#include <stdlib.h>
+#include <string.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <sysevent/sysevent.h>
```
Comment (sadhyama) [inline src/webcfg_wanhandle.c:26]: Remove this sysevent dependency, it will cause build issue in local PC builds

File: src/webcfg_wanhandle.c [hunk 2]
```diff
@@ -18,6 +18,103 @@
...
+            if(newValue == NULL) {
+                    WebcfgError(("NewValue is NULL\n"));
+            }
+            if(oldValue == NULL) {
+                    WebcfgError(("oldValue is NULL\n"));
+            }
+     }
+     if(newValue) {
+         char *CurrentWANStatus = NULL;
+         CurrentWANStatus = (char *) rbusValue_GetString(newValue, NULL);
```
Comment (sadhyama) [inline src/webcfg_wanhandle.c:65]: Add NULL check for CurrentWANStatus before string operation and add an error log if the value is NULL from rbus

File: src/webcfg_wanhandle.c [hunk 3]
```diff
@@ -18,6 +18,103 @@
 
 #include "webcfg_wanhandle.h"
 
+#ifdef _SKY_HUB_COMMON_PRODUCT_REQ_
+#include <stdlib.h>
+#include <string.h>
+#include <pthread.h>
```
[the remark this answers is not in the document]
Comment (PalakshaGowda) [reply src/webcfg_wanhandle.c]: done

File: src/webcfg_wanhandle.c (hunk 2 above)
Comment (PalakshaGowda) [reply src/webcfg_wanhandle.c:63]: done

**Review by sadhyama — APPROVED** [review]

File: src/webcfg_wanhandle.c [hunk 4]
```diff
@@ -18,6 +18,107 @@
...
+        subscribeAsyncHandler,
+        "Webcfg_InterfaceStatus",
+        10*20);
+      if(rc != RBUS_ERROR_SUCCESS) {
+              WebcfgError("%s subscribe failed : %d - %s\n", WEBCFG_WANSTATUS_PARAM, rc, rbusError_ToString(rc));
+      }
+         return rc;
+}
+#endif
+
```
Comment (sadhyama) [inline src/webcfg_wanhandle.c:119]: please check topic verification with this change as it may impact comcast event handler patch

**Pull request merged**